apis/elasticloadbalancing/v1alpha1/elb_types.go

```diff
@@ -0,0 +1,175 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// Tag defines a key value pair that can be attached to an ELB
+type Tag struct {
+
+	// The key of the tag.
+	Key string `json:"key"`
+
+	// The value of the tag.
+	// +optional
+	Value *string `json:"value,omitempty"`
+}
+
+// Listener represents the port binding(s) between the ELB and EC2 instances.
+type Listener struct {
+
+	// The port on which the instance is listening.
+	InstancePort int64 `json:"instancePort"`
+
+	// The protocol to use for routing traffic to instances: HTTP, HTTPS, TCP, or
+	// SSL.
+	// +optional
+	InstanceProtocol *string `json:"instanceProtocol,omitempty"`
+
+	// The port on which the load balancer is listening.
+	LoadBalancerPort int64 `json:"loadBalancerPort"`
+
+	// The load balancer transport protocol to use for routing: HTTP, HTTPS, TCP,
+	// or SSL.
+	Protocol string `json:"protocol"`
+
+	// The Amazon Resource Name (ARN) of the server certificate.
+	// +optional
+	SSLCertificateID *string `json:"sslCertificateId,omitempty"`
+}
+
+// BackendServerDescription provides information about the instances attached to the ELB.
+type BackendServerDescription struct {
+
+	// The port on which the EC2 instance is listening.
+	InstancePort int64 `json:"instancePort,omitempty"`
+
+	// The names of the policies enabled for the EC2 instance.
+	PolicyNames []string `json:"policyNames,omitempty"`
+}
+
+// HealthCheck defines the rules that the ELB uses to decide if an attached instance is healthy.
+type HealthCheck struct {
+
+	// The number of consecutive health checks successes required before moving
+	// the instance to the Healthy state.
+	HealthyThreshold int64 `json:"healthCheck"`
+
+	// The approximate interval, in seconds, between health checks of an individual
+	// instance.
+	Interval int64 `json:"interval"`
+
+	// The instance being checked.
+	Target string `json:"target"`
+
+	// The amount of time, in seconds, during which no response means a failed health
+	// check.
+	Timeout int64 `json:"timeout"`
+
+	// The number of consecutive health check failures required before moving the
+	// instance to the Unhealthy state.
+	UnhealthyThreshold int64 `json:"unhealthyThreshold"`
+}
+
+// ELBParameters define the desired state of an AWS ELB.
+type ELBParameters struct {
+	// One or more Availability Zones from the same region as the load balancer.
+	AvailabilityZones []string `json:"availabilityZones"`
+
+	// Information about the health checks conducted on the load balancer.
+	HealthCheck *HealthCheck `json:"healthCheck,omitempty"`
+
+	// The listeners for this ELB.
+	Listeners []Listener `json:"listeners"`
+
+	// The type of a load balancer. Valid only for load balancers in a VPC.
+	// +optional
+	// +immutable
```

Comment:
I guess you will have to remove this comment

Reply:
AFAIK, a field can be both `optional` and `immutable` at the same time.

Reply:
Yup, saw the use-case as well